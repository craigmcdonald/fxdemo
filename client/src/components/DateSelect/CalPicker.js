import React from 'react'
import PropTypes from 'prop-types'
import DatePicker from 'react-datepicker/lib/datepicker'
import moment from 'moment'
import style from './CalPicker.scss'
import 'react-datepicker/dist/react-datepicker-cssmodules.css'
const { func, object } = PropTypes

const CalPicker = React.createClass({
  propTypes: {
    minDate: object,
    maxDate: object,
    selected: object,
    onSelect: func
  },
  render() {
    return (
      <div styleName='style.cal-picker'>
        <DatePicker
          minDate={moment().subtract(90,'days')}
          maxDate={moment()}
          selected={this.getValue()}
          onSelect={(val => this.setValue(val))}
        />
      </div>
    )
  },
  getValue() {
    return this.props.selected
  },
  setValue(val) {
    this.props.onSelect(val)
  }
})

export default CalPicker
