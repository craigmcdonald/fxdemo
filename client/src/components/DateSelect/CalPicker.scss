.cal-picker {
  width:250px;
  margin-bottom: 15px;
}
.cal-picker input {
  padding-left:10px;
  font-size: 16px;
  font-family: inherit;
  height: 33px;
  width: 238px;
  border: 1px solid #ccc;
  border-radius: 5px;
}
