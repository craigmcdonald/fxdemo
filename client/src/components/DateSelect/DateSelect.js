import React from 'react'
import { FormInput } from 'react-form'
import CalPicker from './CalPicker'
import PropTypes from 'prop-types'
import moment from 'moment'
const { string, object } = PropTypes

const DateSelect = React.createClass({
  propTypes: {
    name: string,
    currentDate: object
  },
  render () {
    var name = this.props.name
    return (
      <div className='fx-form-datepicker-outer'>
        <FormInput field={name}>
          {({getValue, setValue}) => {
            return (
              <div className='fx-form-datepicker-inner'>
                <CalPicker
                  minDate={moment().subtract(90,'days')}
                  maxDate={moment()}
                  selected={getValue()}
                  onSelect={setValue}
                />
              </div>
            )
          }}
        </FormInput>
      </div>
    )
  }
})

export default DateSelect
