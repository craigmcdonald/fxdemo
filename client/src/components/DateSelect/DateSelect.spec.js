import React from 'react'
import ReactDOM from 'react-dom'
import DateSelect from './DateSelect'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'


test('DateSelect Snapshot Test', () => {
  const component = shallow(
    <DateSelect
      name={'foo'}
    />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
