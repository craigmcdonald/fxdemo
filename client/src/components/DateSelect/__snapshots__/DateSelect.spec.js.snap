// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelect Snapshot Test 1`] = `
<div
  className="fx-form-datepicker-outer"
>
  <FormInput
    field="foo"
  />
</div>
`;
